core/templates/pages/exploration-player-page/layout-directives/exploration-footer.component.spec.ts

```diff
@@ -141,6 +167,315 @@ describe('ExplorationFooterComponent', () => {
     expect(component.hintsAndSolutionsAreSupported).toBeFalse();
   });
 
+  it('should push 1 in checkpointArray for the first card', () => {
```

Comment:
Prefer to write descriptions so that it reflects the users perspective avoiding implementation details (e.g. variable names). See  https://github.com/oppia/oppia/wiki/Frontend-tests#name-tests-clearly

Reply:
Done!